### PR TITLE
add more golang linters and address lint issues

### PR DESCRIPTION
**Pull request by hezhizhen**
#### What type of PR is this?

kind/enhancement

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by dixudx** [comment]
@hezhizhen Please adding your golang lint rules as well. This would help make CIs happier.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+run:
+  timeout: 10m
+
+linters:
+  disable-all: true
+  enable:
+    - structcheck
```
Comment (dixudx) [inline .golangci.yml:7]: Prefer to add to https://github.com/clusternet/clusternet/blob/main/Makefile#L53

**Review by dixudx — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -57,7 +57,15 @@ lint: golangci-lint
       -E deadcode \
       -E unused \
       -E varcheck \
-      -E ineffassign
+      -E ineffassign \
```
Comment (dixudx) [inline Makefile:60]: inconsistent indents.

Please correct them. Thanks

**Review by dixudx — COMMENTED** [review]
Excellent work!

File: Makefile (hunk 2 above)
Comment (dixudx) [reply Makefile:60]: Line 61-68 vs Line 60

Comment (hezhizhen) [reply Makefile:60]: PTAL

**Comment by hezhizhen** [comment]
I'll fix other problems in the afternoon.

**Pull request merged**